include/IECore/MeshAlgo.h

```diff
@@ -54,6 +54,8 @@ std::pair<PrimitiveVariable, PrimitiveVariable> calculateTangents( const MeshPri
 	const std::string &position = "P"
 );
 
+PrimitiveVariable resamplePrimitiveVariable( const MeshPrimitive *mesh, const std::string &primitiveVariable, PrimitiveVariable::Interpolation interpolation );
```

Comment:
> I prefer the simpler interface of named primvars and raising exceptions if there it's missing or not the right interpolation.

I'm not entirely sold on this - the caller will still have to deal with the problem somewhere, and it might be better to design the API so they have to deal with it upfront. Taking the ResamplePrimitiveVariables node as an example client, it'll be iterating all primvars anyway to see if their names match the patterns provided by the user, so they'll have done the lookup already, and we'll be repeating it unnecessary in `resample()`.

> Also resamplePrimitiveVariable is responsible for finding the PrimVar on the MeshPrimitive. A caller might get a primVar from another Primitive and attempt to use it in this function and we'd have to guard against this also.

I'm a bit more sold on this argument, but bear in mind the API doesn't enforce the validity of variables (that's why `Primitive::isPrimitiveVariableValid()`) exists.

While we're chatting about this, I realised there's another subtle problem. When the interpolations match, we're returning the original data from the (const) MeshPrimitive, but through an interface that makes it appear that we're getting a fresh resampled copy we can do what we want with. Should we worry about that?